src/com/ansorgit/plugins/bash/lang/parser/ParsingStateData.java

```diff
@@ -38,15 +43,23 @@ public boolean isInSimpleCommand() {
         return inSimpleCommand > 0;
     }
 
-    public void pushHeredocMarker() {
+    public void pushHeredocMarker(int id) {
         heredocMarkers++;
+        heredocMarkersIndexSet.add(id);
     }
 
     public boolean expectsHeredocMarker() {
         return heredocMarkers > 0;
     }
 
     public void popHeredocMarker() {
-        heredocMarkers--;
+        heredocMarkers = heredocMarkers - 1;
```

Comment:
Why not `heredocMarkers--`?

Reply:
Actually, I don't like increment variables like that, but I'm OK to revert